### PR TITLE
model: Add build_environment_variable

### PR DESCRIPTION
**Pull request by richbayliss**
Change-type: patch
Depends-on: https://github.com/balena-io/balena-api/pull/2652
Signed-off-by: Rich Bayliss <rich@balena.io>

### CODE REVIEW

File: typings/balena-sdk.d.ts [hunk 1]
```diff
@@ -317,6 +317,7 @@ declare namespace BalenaSdk {
 		application_environment_variable: ReverseNavigationResource<
 			ApplicationVariable
 		>;
+		build_environment_variable: ReverseNavigationResource<ApplicationVariable>;
```
Comment (thgreasi) [inline typings/balena-sdk.d.ts:320]: Even thought those two are the same, let's create a separate `BuildVariable` type.

File: lib/models/application.js [hunk 2]
```diff
@@ -1650,6 +1661,131 @@ const getApplicationModel = function (deps, opts) {
 			remove: envVarModel.remove,
 		}),
 
+		/**
+		 * @namespace balena.models.application.buildEnvVar
```
Comment (thgreasi) [inline lib/models/application.js:1665]: Not 100% sure about this, but for consistency I think the module's namespace should be `buildVar`.
What do you think?

File: lib/models/application.js [hunk 3]
```diff
@@ -1650,6 +1661,131 @@ const getApplicationModel = function (deps, opts) {
...
+		 * @memberof balena.models.application
+		 */
+		buildEnvVar: addCallbackSupportToModule({
```
Comment (thgreasi) [inline lib/models/application.js:1668]: Don't forget to add typings in the `balena-sdk.d.ts` for the new model.

File: typings/balena-sdk.d.ts (hunk 1 above)
Comment (richbayliss) [reply typings/balena-sdk.d.ts:320]: No probs

File: lib/models/application.js (hunk 2 above)
Comment (richbayliss) [reply lib/models/application.js:1665]: I thought to keep to the API model name *build*_*env*ironment_*var*iable. If you'd rather go with `buildVar` then I am not going to fight it 👍

Comment (thgreasi) [reply lib/models/application.js]: I'm fine with either :

**Pull request merged**